doc: Documentation fixes (#14226)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

      Set to ``TLS_PEER_VERIFY_REQUIRED`` by default.
    * Set the ``TLS_HOSTNAME`` socket option to ``NULL`` to disable the hostname verification.

>   * The application can now be built on nRF9160 DK boards with revision older than 0.14.0.

Reviewer: ```suggestion
  * You can now build the application on nRF9160 DK boards with revision older than 0.14.0.
```

      Set to ``TLS_PEER_VERIFY_REQUIRED`` by default.
    * Set the ``TLS_HOSTNAME`` socket option to ``NULL`` to disable the hostname verification.

>   * You can now build the application on nRF9160 DK boards with revision older than 0.14.0.

Reviewer 2: ```suggestion
  * You can now build the application on nRF9160 DK boards with revisions older than 0.14.0.
```

    * The ``family`` parameter to the :c:struct:`download_client_cfg` structure.
      This is used to optimize the download sequence when the device only support IPv4 or IPv6.

  * Updated:

    * IPv6 support changed from compile time to runtime, and is default enabled.
    * IPv6 to IPv4 fallback is done when both DNS request and TCP/TLS connect fails.
    * HTTP downloads forward data fragments to a callback only when the buffer is full.

>   * Removed the :kconfig:option:`CONFIG_DOWNLOAD_CLIENT_IPV6` Kconfig option.

Reviewer: ```suggestion
  * Removed the ``CONFIG_DOWNLOAD_CLIENT_IPV6`` Kconfig option.
```
If it is removed, I'd guess it is no longer available in the Kconfig reference.
